jetty-server/src/test/java/org/eclipse/jetty/server/GracefulStopTest.java

```diff
@@ -55,6 +63,136 @@
 @RunWith(AdvancedRunner.class)
 public class GracefulStopTest 
 {
+    public void testSlowClose(long stopTimeout, long closeWait, Matcher<Long> stopTimeMatcher) throws Exception
+    {
+        Server server= new Server();
+        server.setStopTimeout(stopTimeout);
+
+        CountDownLatch closed = new CountDownLatch(1);
+        ServerConnector connector = new ServerConnector(server, new HttpConnectionFactory() 
+        {
+
+            @Override
+            public Connection newConnection(Connector connector, EndPoint endPoint)
+            {
+                // Slow closing connection
+                HttpConnection conn = new HttpConnection(getHttpConfiguration(), connector, endPoint, getHttpCompliance(), isRecordHttpComplianceViolations())
+                {
+                    @Override
+                    public void close()
+                    {
+                        try
+                        {
+                            Thread.sleep(closeWait);
+                            super.close();
+                        }
+                        catch(Exception e)
+                        {
+                            // e.printStackTrace();
+                        }
+                        finally
+                        {
+                            closed.countDown();
+                        }
+                    }
+                };
+                return configure(conn, connector, endPoint);
+            }
+            
+        });
+        connector.setPort(0);
+        server.addConnector(connector);
+
+        NoopHandler handler = new NoopHandler();
+        server.setHandler(handler);
+
+        server.start();
+        final int port=connector.getLocalPort();
+        Socket client = new Socket("127.0.0.1", port);
+        client.setSoTimeout(10000);
+        client.getOutputStream().write((
+                "GET / HTTP/1.0\r\n"+
```

Comment:
Wouldn't using HTTP/1.0 cause a race to close the connection on the server, since the dispatched thread will try to close it while the call to `server.stop()` below also will try to close it ?
I think it's better to use HTTP/1.1, since you only look for the end of headers below.
Calling `server.stop()` will trigger the close of the connection, and will be the only trigger.